Generals/Code/GameEngine/Source/Common/System/StackDump.cpp

```diff
@@ -76,6 +76,7 @@ void StackDump(void (*callback)(const char*))
 
 	DWORD myeip,myesp,myebp;
 
+#ifdef _MSC_VER
```

Comment:
I think it is better to make this VS6 build specific.

```
#if defined(_MSC_VER) && _MSC_VER < 1300
```

Same for the various other ifdef's in this change.

Reply:
Probably true, I was just being conservative. This code will need to be properly tested, but I am now not in position to do so.. :/

Btw doc for `RtlCaptureContext` [says](https://learn.microsoft.com/en-us/windows-hardware/drivers/ddi/ntifs/nf-ntifs-rtlcapturecontext):
> Retrieves a context record in the context of the caller.

I have done some experiments and it seems that caller here means not caller of the `RtlCaptureContext`, but caller of function, which called `RtlCaptureContext`. So maybe context will be one frame off compared to asm. If that is the case, either skip value would need to be adjusted by one or dummy wrapper function would need to be created such as:
```c++
static void CaptureContextWrapper(CONTEXT ctx*) {
    RtlCaptureContext(ctx);
}
```

Reply:
I tested this also with MSVC, seems that `RtlCaptureContext` really captures context one frame higher then, where it was called (see my [experiment](https://godbolt.org/z/hf9fajP9v)).

I think placing `RtlCaptureContext`  in dummy wrapper function will probably be easiest and least invasive solution to match the asm behavior.